components/places/src/storage.rs

```diff
@@ -459,6 +459,17 @@ mod tests {
         assert_eq!(pi.page.last_visit_date_local, early_time.into());
         assert_eq!(pi.page.visit_count_remote, 1);
         assert_eq!(pi.page.last_visit_date_remote, late_time.into());
+
+        // Delete all visits.
+        conn.execute_named_cached(&sql, &[(":row_id", &rid2)]).expect("delete should work");
+        conn.execute_named_cached(&sql, &[(":row_id", &rid4)]).expect("delete should work");
+        // It may turn out that we also delete the place after deleting all
+        // visits, but for now we don't - check the values are sane though.
```

Comment:
(Unless it's bookmarked, I think that's why we have `foreign_count` on Desktop).